src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java

```diff
@@ -0,0 +1,154 @@
+    fileNames.add(TEST_TEMPLATE_FILE);
+    // TODO(shinfan): Add gRPC service mock template.
+    return fileNames;
+  }
+
+  private ModelTypeTable createTypeTable() {
+    return new ModelTypeTable(new JavaTypeTable(), new JavaModelTypeNameConverter());
+  }
+
+  private void addImports(SurfaceTransformerContext context) {
+    ModelTypeTable typeTable = context.getTypeTable();
+    typeTable.saveNicknameFor("org.junit.After");
+    typeTable.saveNicknameFor("org.junit.AfterClass");
+    typeTable.saveNicknameFor("org.junit.Before");
+    typeTable.saveNicknameFor("org.junit.BeforeClass");
+    typeTable.saveNicknameFor("org.junit.Test");
+    typeTable.saveNicknameFor("java.io.IOException");
+    typeTable.saveNicknameFor("java.util.List");
+    typeTable.saveNicknameFor("com.google.api.gax.testing.ValueGenerator");
+    typeTable.saveNicknameFor("com.google.api.gax.testing.LocalServiceHelper");
+    typeTable.saveNicknameFor("com.google.protobuf.GeneratedMessage");
+    typeTable.saveNicknameFor("junit.framework.Assert");
+  }
+
+  private GapicSurfaceTestClassView createTestClassView(SurfaceTransformerContext context) {
+    Interface service = context.getInterface();
+    String outputPath = pathMapper.getOutputPath(service, context.getApiConfig());
+
+    GapicSurfaceTestClassView testClass =
+        GapicSurfaceTestClassView.newBuilder()
+            .packageName(context.getApiConfig().getPackageName())
+            .apiSettingsClassName(context.getNamer().getApiSettingsClassName(service))
+            .apiClassName(context.getNamer().getApiWrapperClassName(service))
+            .name(context.getNamer().getTestClassName(service))
+            .mockServiceClassName(context.getNamer().getMockServiceClassName(service))
+            .testCases(createTestCaseViews(context))
+            .outputPath(outputPath)
+            .imports(context.getTypeTable().getImports())
+            .templateFileName(TEST_TEMPLATE_FILE)
+            .build();
+    return testClass;
+  }
+
+  private List<GapicSurfaceTestCaseView> createTestCaseViews(SurfaceTransformerContext context) {
+    ArrayList<GapicSurfaceTestCaseView> testCaseViews = new ArrayList<>();
+    for (Method method : context.getInterface().getMethods()) {
+      MethodTransformerContext methodContext = context.asMethodContext(method);
+      testCaseViews.add(createTestCaseView(methodContext));
+    }
+    return testCaseViews;
+  }
+
+  private GapicSurfaceTestCaseView createTestCaseView(
+      MethodTransformerContext methodContext) {
+    MethodConfig methodConfig = methodContext.getMethodConfig();
+    InitCodeTransformer initCodeTransformer = new InitCodeTransformer();
+    InitCodeView initCodeView =
+        initCodeTransformer.generateInitCode(methodContext, methodConfig.getRequiredFields());
+    List<GapicSurfaceTestAssertView> assertViews =
+        initCodeTransformer.generateTestAssertViews(
+            methodContext, methodConfig.getRequiredFields());
```

Comment:
I feel like it would be nice to not repeat the work of generating the init code (done by both generateInitCode and generateTestAssertViews). Don't make stuff hacky to accomplish it, though.